src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -76,31 +104,29 @@ public static class EndpointOptionsExtensions
 
         internal static IPEndPoint GetPublicSiloEndpoint(this EndpointOptions options)
         {
-            return new IPEndPoint(options.AdvertisedIPAddress, options.SiloPort);
+            return options?.AdvertisedIPAddress != null
```

Comment:
I don't think this check is needed since the validator should throw before that.

We could also throw an exception here just in case, currently the silo cannot work without setting the `AdvertisedIPAddress`. Same comment for `GetPublicProxyEndpoint`